cou/utils/openstack.py

```diff
@@ -147,3 +209,117 @@ def lookup(cls, component: str, version: str) -> List[str]:
+    alphanumerically ordered.  e.g. OpenStack or Ubuntu releases AFTER the
+    z-wrap.
+    """
+
+    openstack_codenames = list(OPENSTACK_CODENAMES.keys())
+    openstack_release_year = list(OPENSTACK_CODENAMES.values())
+
+    def __init__(self, os_release: str):
+        """Initialize the OpenStackRelease object.
+
+        :param os_release: OpenStack release codename.
+        :type os_release: str
+        :raises ValueError: Raises ValueError if OpenStack codename is unknown.
+        """
+        self.os_release = os_release
+
+    def __eq__(self, other):
+        """Do equals."""
+        if not isinstance(other, (str, OpenStackRelease)):
+            return NotImplemented
+        return self.index == self.openstack_codenames.index(other)
+
+    def __ne__(self, other):
+        """Do not equals."""
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        """Do less than."""
+        if not isinstance(other, (str, OpenStackRelease)):
+            return NotImplemented
+        return self.index < self.openstack_codenames.index(other)
+
+    def __ge__(self, other):
+        """Do greater than or equal."""
+        return not self.__lt__(other)
+
+    def __gt__(self, other):
+        """Do greater than."""
+        if not isinstance(other, (str, OpenStackRelease)):
+            return NotImplemented
+        return self.index > self.openstack_codenames.index(other)
+
+    def __le__(self, other):
+        """Do less than or equals."""
+        return not self.__gt__(other)
+
+    def __repr__(self):
+        """Return the representation of CompareOpenStack."""
+        return f"{self.__class__.__name__}<{self.os_release}>"
+
+    @property
+    def os_release(self) -> str:
+        """Return the next OpenStack release codename.
+
+        :return: OpenStack release codename.
+        :rtype: str
+        """
+        return self._os_release
+
+    @os_release.setter
+    def os_release(self, value: str):
+        """Setter of OpenStack release codename.
+
+        :param value: OpenStack release codename.
+        :type value: str
+        :raises ValueError: Raise ValueError if codename is unknown.
+        """
+        if value not in self.openstack_codenames:
+            raise ValueError(f"OpenStack '{value}' is not in '{self.openstack_codenames}'")
+        self._os_release = value
+        self.index = self.openstack_codenames.index(value)
+
+    @property
+    def next_release(self) -> str:
+        """Return the next OpenStack release codename.
+
+        :return: OpenStack release codename.
+        :rtype: str
+        """
+        try:
+            os_release = self.openstack_codenames[self.index + 1]
+        except IndexError:
+            logger.warning("There is no OpenStack release after %s", self.os_release)
+            os_release = self.openstack_codenames[self.index]
+        return os_release
+
+    @property
+    def release_year(self) -> str:
```

Comment:
in the same spirit as above, the `release_` part is redundant. And `year` is not really correct when release numbers contain months as well.
```suggestion
    def date(self) -> str:
```